HW5.py

```diff
@@ -0,0 +1,586 @@
+    )
+    con_cat_correlation_df_sorted = con_cat_correlation_df_sorted.style.format(
+        {"Predictor 1 Plot": fun, "Predictor 2 Plot": fun}
+    )
+    con_cat_table = con_cat_correlation_df_sorted.render()
+    text_file = open("con_cat_correlation_table.html", "w")
+    text_file.write(con_cat_table)
+
+    # BUILD CAT/CAT CORRELATION TABLE
+    cat_cat_correlation_df = df[df["Predictor 1 Type"] == "cat"]
+    cat_cat_correlation_df = cat_cat_correlation_df[
+        cat_cat_correlation_df["Predictor 2 Type"] == "cat"
+    ]
+    cat_cat_correlation_df_sorted = cat_cat_correlation_df.sort_values(
+        "Correlation Coefficient", ascending=False
+    )
+    cat_cat_correlation_df_sorted = cat_cat_correlation_df_sorted.style.format(
+        {"Predictor 1 Plot": fun, "Predictor 2 Plot": fun}
+    )
+    cat_cat_table = cat_cat_correlation_df_sorted.render()
+    text_file = open("cat_cat_correlation_table.html", "w")
+    text_file.write(cat_cat_table)
+
+    # CORRELATION MATRICES
+    pred_types = [(pred_name, classify_type(X[pred_name])) for pred_name in predictors]
+    con_predictors = [x for (x, y) in pred_types if y == "con"]
+    con_data = X[con_predictors]
+    cat_predictors = [x for (x, y) in pred_types if y == "cat"]
+    cat_data = X[cat_predictors]
+
+    # cat/con
+    correlation_matrix = X.corr()
+    cat_con_fig = px.imshow(correlation_matrix)
+    cat_con_fig.write_html(
+        file="plots/cat_con_correlation_matrix_plot.html",
+        include_plotlyjs="cdn",
+        include_mathjax="cdn",
+    )
+
+    # con/con
+    correlation_matrix_con = con_data.corr()
+    con_con_fig = px.imshow(correlation_matrix_con)
+    con_con_fig.write_html(
+        file="plots/con_con_correlation_matrix_plot.html",
+        include_plotlyjs="cdn",
+        include_mathjax="cdn",
+    )
+
+    # cat/cat
+    correlation_matrix_cat = cat_data.corr()
+    cat_cat_fig = px.imshow(correlation_matrix_cat)
+    cat_cat_fig.write_html(
+        file="plots/cat_cat_correlation_matrix_plot.html",
+        include_plotlyjs="cdn",
+        include_mathjax="cdn",
+    )
+
+    correlation_matrix_df = pd.DataFrame(
+        list(
+            zip(
+                ["plots/con_con_correlation_matrix_plot.html"],
+                ["plots/con_con_correlation_matrix_plot.html"],
+                ["plots/cat_cat_correlation_matrix_plot.html"],
+            )
+        ),
+        columns=[
+            "Cat/Con Correlation Matrix",
+            "Con/Con Correlation Matrix",
+            "Cat/Cat Correlation Matrix",
+        ],
+    )
+    correlation_matrix_df_styled = correlation_matrix_df.style.format(
+        {
+            "Cat/Con Correlation Matrix": fun,
+            "Con/Con Correlation Matrix": fun,
+            "Cat/Cat Correlation Matrix": fun,
+        }
+    )
+    correlation_matrix_table = correlation_matrix_df_styled.render()
+    text_file = open("correlation_matrices.html", "w")
+    text_file.write(correlation_matrix_table)
```

Comment:
1. Maybe adding a `tables/` directory would keep things a bit neater?  As things stand, correlation plots are getting saved in the root and brute force plots are getting saved in a separate folder.
2. It is a bit inconvenient to have to pull up the reports in browser.  [This native Python module](https://docs.python.org/3/library/webbrowser.html) can make things a bit easier to test since it can automatically open a browser to the correct directory.